### DISCUSSION TITLE
Failed to Connect - Compact P Kol Sol Polar XL HMI Air 9+

### DISCUSSION THREAD

**Discussion by k1ssj6n0s** [veista/nilan#156] [Q&A]
Hello!

I tried to connect a Compact P + Air 9 with the integration, but I only get the message "Failed to Connect".
I'm trying to connect with a Waveshare USB TO RS485 device, but I've tried a Waveshare RS485 TO ETH , unfortunately that didn't work either.

Nilan integration: 1.2.16 and 1.2.17

Best regards

[home-assistant_modbus_2024-11-03T21-26-40.835Z.log](https://github.com/user-attachments/files/17612224/home-assistant_modbus_2024-11-03T21-26-40.835Z.log)
![IMG_20241025_194000](https://github.com/user-attachments/assets/820793a5-c5f5-49a6-bcd2-48fcfa97f5d2)
![IMG_20241025_194116](https://github.com/user-attachments/assets/749cc45f-ad4c-4207-ad32-912f1e4fa281)
![IMG_20241025_194015](https://github.com/user-attachments/assets/d4326596-9a8c-49bc-9346-be71389e5b48)
![IMG_20241103_225800](https://github.com/user-attachments/assets/95315b1a-95d3-48a8-bcf3-bf95afea015b)

**Comment by veista** [comment]
Usb to rs485 should be simple and bulletproof.

1. Check wiring. See wiki. A, B, Ground
2. Check Usb path.
3. Check Modbus ID

**Reply by k1ssj6n0s** [reply]
Hello,

Thank you for your reply.

Unfortunately I have already tried all of these, but now I have checked everything again, and I have also tried the wiring by swapping A and B. (I also did a continuity test between the two ends.)

Unfortunately it still doesn't work.

What is interesting for me is that on the top of the device, between the connector and the Nilan gateway, there is a grey box: from the connector on the top of the device, the cable goes to this board in the grey box, and from here another cable goes to the modbus port of the gateway.

I have not found any information or reference to this box in any Nilan documentation.
Is it possible that this is causing the problem?
The box is not connected to anything other than the modbus cables and the power outlet.

Best regards,
Janos
![IMG_20241106_184218](https://github.com/user-attachments/assets/483bb579-d937-402a-8bd0-4450540a2916)
![IMG_20241106_183935](https://github.com/user-attachments/assets/05eccc9f-6f8f-4a95-aaf7-3f1a32f66b24)
[home-assistant_modbus_2024-11-06T22-05-48.137Z.log](https://github.com/user-attachments/files/17653775/home-assistant_modbus_2024-11-06T22-05-48.137Z.log)

**Reply by veista** [reply]
I do not understand your setup, but you should not have the nilan gateway connected at all. This integration does not work with the gateway.

**Reply by veista** [reply]
https://github.com/veista/nilan/wiki/Wiring-for-CTS602

**Reply by veista** [reply]
You should replace the gateway with the usb to rs485 cable or bridge device. If you have doubts I highly recommend to ask Nilan for help. Atleast for me they are extremely helpful always.

**Reply by k1ssj6n0s** [reply]
I configured the USB to RS485 adapter according to the wiring, so when I tried to use it, I disconnected the devices shown in the previous pictures - the grey box and the gateway.

Since all the documentation had the gateway directly after the connector at the end of the cable, I figured that I had something different in the configuration that could be causing the connection error.

But in that case I will contact Nilan.

Thanks for your work and help!

**Reply by k1ssj6n0s** [reply]
Just one more question.

The grey box has a CTS602 board in it based on my research,
So I thought of disconnecting only the gateway but leaving the grey box connected.

I tried the modbus connection again and although i now get a similar error, the logs now contain much more information.

Could you please see if this shows any useful information?

[home-assistant_modbus_2024-11-06T23-38-10.018Z.log](https://github.com/user-attachments/files/17654268/home-assistant_modbus_2024-11-06T23-38-10.018Z.log)

**Reply by veista** [reply]
Could you please give me a log of only this integration.

**Reply by k1ssj6n0s** [reply]
I finally managed to connect to the device.
The WAGO on the top of the device connects to the A2, B2 pin of the CTS602 CN7 in the grey box mentioned earlier: as described, I tried to connect the Waveshare USB adapter to this WAGO.
(Interestingly, the GND is not connected between the WAGO and the CTS602, although it seems that it was intended to be connected, but the two ends of the cable used a different wire.)
So the connection did not work.

However, when connected to the CN7 connector A1, B1, GND of the CTS602 with the Waveshare USB adapter, the connection works.

I don't know the best way to extract the logs related to the integration, so I parsed a few log into the attached nilan.log.

[nilan.log](https://github.com/user-attachments/files/17690715/nilan.log)

**Reply by veista** [reply]
Does it work now?

**Reply by k1ssj6n0s** [reply]
Yes, it basically works.

According to another forum post, since only Modbus v2 is available, it doesn't support more options.
(At least I think that's why there are a lot of sensors, but not all options are HA modifiable.)

**Reply by veista** [reply]
That does not apply to compact air/geo models. You should have all functionality.

**Reply by veista** [reply]
Can you share screenshots of how it looks on the device page?

**Reply by k1ssj6n0s** [reply]
I attached, although not all item names are shown in full width.

I will try again to compare the options available in the device menu, in the Nilan app and in HA, because I might not find everything just because of the different name usage.
But I think even this state might be appropriate, I didn't plan to configure all parameters through ModBus.

![Screenshot 2024-11-11 141513](https://github.com/user-attachments/assets/6308c0fd-32eb-4991-b628-169b79380864)
![Screenshot 2024-11-11 141554](https://github.com/user-attachments/assets/cf756c23-e4a4-4153-bdff-fe0bc71b9684)
![Screenshot 2024-11-11 141621](https://github.com/user-attachments/assets/21609f36-d073-4a5d-bed0-436d2b5a14b0)
![Screenshot 2024-11-11 141648](https://github.com/user-attachments/assets/8cbb0155-cbbf-48cc-bf40-46e95460011f)
![Screenshot 2024-11-11 141709](https://github.com/user-attachments/assets/cba45a86-30ec-4313-a671-016d399b06f0)
![Screenshot 2024-11-11 141732](https://github.com/user-attachments/assets/c5eb7ce2-f368-45bd-b042-5bf19fb86914)
![Screenshot 2024-11-11 141753](https://github.com/user-attachments/assets/f5eb829c-ec0c-4854-8814-21a69c85dad0)

**Reply by veista** [reply]
Thanks. Seems to be working as should!

**Reply by k1ssj6n0s** [reply]
Thanks for the confirmation!
